### DISCUSSION TITLE
Using IDS to Validate Property Attributes

### DISCUSSION THREAD

**Discussion by bimintelligence** [buildingSMART/IDS#372] [Q&A]
Hello everyone,

I’m wondering if it’s possible to validate property attributes using IDS.

For example, I’d like to check the attribute “People” within the Pset_SpaceThermalLoad of an IfcBuilding, which is expressed using IfcPropertyBoundedValue.

If this is feasible, I’d also like to know how the XML structure can be used to verify the UpperBoundValue and LowerBoundValue.

Thank you in advance for any guidance!

**Comment by atomczak** [comment]
Look at the examples of IDS and IFC files using IfcPropertyBoundedValue here:
https://github.com/search?q=repo%3AbuildingSMART%2FIDS%20path%3A%2F%5EDocumentation%5C%2FImplementersDocumentation%5C%2FTestCases%5C%2Fproperty%5C%2F%2F%20any_matching_value_in_a_bounded_property_will_pass&type=code

However, I'm not sure they are fully correct, so I created this issue: https://github.com/buildingSMART/IDS/issues/371

**Reply by bimintelligence** [reply]
Thank you for pointing me to those examples. 
Yes, I did take a look, but it seems that the simpleValue representation of "Foo" doesn’t fully capture the three distinct values within an IFCPROPERTYBOUNDEDVALUE. 
I'm still a bit confused about how to properly express this.
![image](https://github.com/user-attachments/assets/10cd7c50-d637-4d45-a948-67c39b9fcb84)
![image](https://github.com/user-attachments/assets/66b0fb2c-0c47-4d6c-ad3c-3e5095a2234b)

**Reply by atomczak** [reply]
what do you want to check? "'Foo' lower and upper values should be exactly 1234 and 4321"? You could use IDS restriction minInclusive/maxInclusive to narrow down the range. If the bounds in IFC are within the bounds of IDS, that should pass. I think so, at least :)

**Reply by bimintelligence** [reply]
The test cases for IDS restrictions mostly focus on a single attribute within a property set, like the ThermalTransmittance in Pset_PlateCommon in the example below.

My confusion lies in two areas:

Which attribute of Foo is being restricted by simpleValue—is it UpperBoundValue, LowerBoundValue, or SetPointValue?
How can I narrow down an attribute within an attribute in a property set?
In the example above, Foo itself is already an attribute of Foo_Bar. When it comes to further narrowing down something like UpperBoundValue, it seems there are no test cases to clarify how to generate IDS for this scenario.

![image](https://github.com/user-attachments/assets/977da45d-3dba-48c3-84a7-eb63e1312877)
